Add release notes for 0.218 (#12517)
prestodb/presto: presto-docs/src/main/sphinx/release/release-0.218.rst

=============
Release 0.218
=============

General Changes
---------------

> * Add support for using binary (smile) encoding for coordinator-to-worker communication.

Reviewer: Reformatting a bit:

```
* Add support for using binary (smile) encoding for coordinator-to-worker communication.
  This feature is experimental, and it can be enabled with the ``experimental.internal-communication.binary-transport-enabled`` config option.
  Enabling this feature should help with coordinator scalability, and should reduce network, cpu, and memory usage on the coordinator.
```

Reviewer: Additions go to the bottom of the release note items per https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines.

This feature is experimental, and it can be enabled with the ``experimental.internal-communication.binary-transport-enabled`` config option.
Enabling this feature should help with coordinator scalability, and should reduce network, cpu, and memory usage on the coordinator.

> * Add ST_Area for SphericalGeography

Reviewer: - Need `.` at the end. There are other items below that need `.` at the end.
- Need to link to the documentation of `ST_Area`. 
- What's is `SphericalGeography`? It's vague for the reader, so we should rephrase this.

Reviewer: Additions go to the bottom of the release note items per https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines.

> * Add a new system table ``system.metadata.analyze_properties`` to display available

Reviewer: ```
* Add a new system table ``system.metadata.analyze_properties`` that stores the
  properties that can be used with the ``ANALYZE`` statement.
```

Reviewer: Additions go to the bottom of the release note items per https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines.

properties that can be used in ``ANALYZE`` statement.

> * Optimize performance for queries with COALESCE join keys on top of FULL OUTER JOIN

Reviewer 2: Improve performance for certain queries with ``COALESCE`` join keys on top of ``FULL OUTER JOIN``

Reviewer: - Need `.` at the end.
- ``COALESCE`` join keys -> I think this should be rephrased as not clear to me what is meant here.

> * Fix failures in regular expression functions for certain inputs when the pattern contained word boundaries.

Reviewer: I think we should rephrase `for certain inputs when the pattern contained word boundaries.`  as I didn't understand the condition. Is it like the given pattern matches word boundaries or sth else?

* Consider estimated query peak memory when making cost based decisions.

> * Add an optional additional argument to split_to_map to resolve key conflicts

Reviewer: - `.` at the end.
- Need to link to `split_to_map` doc.
- Add double backticks around split_to_map.

> * Improve performance for queries with an outer join followed by `LIMIT`.

Reviewer: In some release notes I see we use 

```
``OUTER JOIN``
```

> * Fix a ``ClassLoader`` crashing issue for plugins providing ``EventListenerFactory``.

Reviewer: - What does class loader crashing mean here? We need to be more specific here.
- `plugins providing an`

* Add support for role management (see :doc:`/sql/create-role`).  Note, using :doc:`/sql/set-role`

>   requires an up-to-date client library. (:issue:`90`)

Reviewer: - `Please note that, using ...`
- Which client library are we talking about here?

Security Changes
----------------

* Make principal hostname configurable in Kerberos authenticator.

Hive Connector Changes
----------------------

* Add support for skipping Glacier files in PrestoS3FileSystem with the ``hive.s3.skip-glacier-objects`` config option, which is false by default.
* Support Parquet files written with Parquet 1.9+ that use ``DELTA_BINARY_PACKED`` encoding with the Parquet ``INT64`` type.
* Support dictionary filtering for Parquet v2 files using `RLE_DICTIONARY` encoding.

> * Add option to preload all splits with ``hive.preload-splits-for-grouped-execution`` config option, which is false by default. Turning this on might slightly increase query latency for grouped execution.

Reviewer: - `might` -> `can`
-  Should we say: `the query latency with grouped execution`

* Improve Parquet reader performance by reducing the number of footer reads.

Elasticsearch Connector Changes

> ----------------------------------

Reviewer: Too many `-` characters here, you should remove 3 of them.

> * Add support for Search Guard in Elasticsearch connector. To use this feature the certificate format should be configured with the ``searchguard.ssl.transport.certificate_format`` config property, and the corresponding key path/password should also be configured for the specified certificate format.

Reviewer: Add new lines as this is a single long line:

```
* Add support for Search Guard in Elasticsearch connector. To use this feature the certificate format should be configured with the 
  ``searchguard.ssl.transport.certificate_format`` config property, and the corresponding key path/password should also be configured 
  for the specified certificate format.
```

=============
Release 0.218
=============

General Changes
---------------

> * Fix failures in regular expression functions for certain inputs where the pattern contains word boundaries (i.e. ``\b``).

Reviewer 3: ```suggestion
* Fix failures in regular expression functions for certain inputs where the pattern contains word boundaries (e.g. ``\b``).
```

> * Fix an issue that may cause the Presto process to crash when using plugins that provide an event listener. (:issue:`11951`)

Reviewer 3: ```suggestion
* Fix an issue that may cause a crash when using plugins that provide an event listener. (:issue:`11951`)
```

* Add a system table ``system.metadata.analyze_properties`` that shows the properties supported by the ``ANALYZE`` statement.
* Add support for resolving key conflicts when using :func:`split_to_map`.
* Add support for role management (see :doc:`/sql/create-role`). Client library version 0.218 is required to use :doc:`/sql/set-role`. (:issue:`11645`)
* Add support for processing JSON protocol messages by generating bytecode on the coordinator.
  This feature is experimental, and it can be enabled with the ``experimental.json-serde-codegen-enabled`` configuration property.

Security Changes
----------------

* Change principal hostname to be configurable in Kerberos authenticator.

Hive Connector Changes
----------------------

* Improve Parquet reader performance by reducing redundant footer reads.
* Add support for skipping Glacier files in Amazon S3. This feature can be enabled by setting the ``hive.s3.skip-glacier-objects`` configuration property.
* Add support for Parquet files written with Parquet v1.9+ that use ``DELTA_BINARY_PACKED`` encoding with the ``INT64`` type.
* Add support for dictionary filtering for Parquet v2 files that use ``RLE_DICTIONARY`` encoding.

> 

Reviewer 3: Why double newlines here and not for other section breaks?

=============
Release 0.218
=============

General Changes
---------------

> * Fix failures in regular expression functions for certain inputs where the pattern contains word boundaries (e.g. ``\b``).

Reviewer: nit: `e.g.,`
